Zend/tests/type_declarations/typed_class_constants_002.phpt

```diff
@@ -0,0 +1,10 @@
+--TEST--
+Typed class constants (type mismatch; compile-time)
+--FILE--
+<?php
+class A {
+	public const string A = 1;
+}
+?>
+--EXPECTF--
+Fatal error: Cannot use int as value for class constant A::A of type string in %s on line %d
```

Comment:
In my opinion, `class` in `class constant` part is superfluous. At least I removed them in https://github.com/php/php-src/pull/5590. (I'm ok however if we decide that `class constant` is necessary, but then I'll update my PR)

Reply:
TBH I don't really like this message. I'd probably prefer something similar to:
`Constant A::A must have a value of type string, int used`
but I'm sure it's still not the best :)

Reply:
@kocsismate I was reusing the message that is used for default value validation in parameters and properties.

Reply:
Thanks! I didn't notice that you use that format. Then I'll try to come up with a format for params/properties separately from this RFC :)

Reply:
> Thanks! I didn't notice that you use that format. Then I'll try to come up with a format for params/properties separately from this RFC :)

...I just dropped the "default" word from the error message since it didn't make much sense in this context.